ivy_models/bert/bert.py

```diff
@@ -0,0 +1,353 @@
+            use_cache = use_cache if use_cache is not None else self.config.use_cache
+        else:
+            use_cache = False
+
+        # past_key_values_length
+        past_key_values_length = (
+            past_key_values[0][0].shape[2] if past_key_values is not None else 0
+        )
+        embeddings = self.embeddings(
+            input_ids,
+            token_type_ids,
+            position_ids,
+            past_key_values_length,
+        )
+
+        encoder_outs = self.encoder(
+            embeddings,
+            attention_mask,
+            encoder_hidden_states,
+            encoder_attention_mask,
+            past_key_values,
+            use_cache,
+            output_attentions,
+        )
+        if self.pooler_out:
+            pooler_out = self.pooler(encoder_outs[0])
+        else:
+            pooler_out = None
+        return {
+            "pooled_output": pooler_out,
+            "last_hidden_state": encoder_outs[0],
+            "attention_probs": encoder_outs[1],
+            "next_decoder_cache": encoder_outs[2],
+        }
+
+
+# Mapping and loading section
+
+
+def custom_map(name):
+    key_map = [
+        ("__v0__", "__0__"),
+        ("__v1__", "__1__"),
+        ("__v10__", "__2__"),
+        ("__v11__", "__3__"),
+    ]
+    key_map = key_map + [
+        (f"__v{i}", f".{j}") for i, j in zip(range(2, 10), range(4, 12))
+    ]
+    key_map = key_map + [
+        ("attention__dense", "attention.output.dense"),
+        ("attention__LayerNorm", "attention.output.LayerNorm"),
+    ]
+    key_map = key_map + [
+        ("ffd__dense1", "intermediate.dense"),
+        ("ffd__dense2", "output.dense"),
+        ("ffd__LayerNorm", "output.LayerNorm"),
+    ]
+    name = name.replace("__w", ".weight").replace("__b", ".bias")
+    name = (
+        name.replace("biasias", "bias")
+        .replace("weighteight", "weight")
+        .replace(".weightord", ".word")
+    )
+    for ref, new in key_map:
+        name = name.replace(ref, new)
+    name = name.replace("__", ".")
+    return name
+
+
+def get_idx_from_map(module_list, name):
+    names = ["v0", "v1", "v10", "v11"] + [
+        f"v{i}" for i in range(2, len(module_list) - 2)
+    ]
+    mapping = dict(zip(names, range(len(names))))
+    return mapping[name]
+
+
+def unflatten_set_module(
+    module,
+    flattened_name,
+    to_set,
+    split_on="__",
+):
+    splits = flattened_name.split(split_on)
+    cont = module
+    for idx, sp in enumerate(splits[:-1]):
+        cont = getattr(cont, sp)
+        if isinstance(cont, list):  # map the list structure to indices
+            mapped_idx = get_idx_from_map(cont, splits[idx + 1])
+            cont = cont[mapped_idx]
+            for s in splits[idx + 2 : -1]:
+                cont = getattr(cont, s)
+            break
+    cont = getattr(cont, "v")  # set the parameter variable to the wanted value
+    setattr(cont, splits[-1], to_set)
+
+
+def load_transformers_weights(
```

Comment:
this seems like a weights helper to me, would you mind to refactor this (and the helper function it is using) into the `weights_helpers.py` file in the `helpers` folder? thanks!

Reply:
would you mind to please reference the existing weights loading helpers and refactor this so that it is returning the weights only instead of a model to be consistent? thanks!

Reply:
I have tried to reinitialize the weights with the new mapped weights but it didn't work for me.

Reply:
I managed to solve the problem